packages/engine/lib/memory/src/shared_memory/mod.rs

```diff
@@ -6,6 +6,8 @@ mod markers;
 mod metaversion;
 mod ptr;
 mod segment;
+// reason: will be removed in a follow-up task
+#[allow(clippy::module_inception)]
 mod shared_memory;
```

Comment:
I think we can probably just move the contents of this into this `mod.rs` file but I'm fine with fixing it in the follow-up

Reply:
The content will be moved into `segment`, thus I'd keep it.